llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2821,6 +2821,47 @@ SDValue DAGCombiner::visitADDLike(SDNode *N) {
   return SDValue();
 }
 
+// Attempt to form avgflooru(A, B) from add(and(A, B), lshr(xor(A, B), 1))
+static SDValue combineFixedwidthToAVGFLOORU(SDNode *N, SelectionDAG &DAG) {
+  assert(N->getOpcode() == ISD::ADD and "ADD node is required here");
```

Comment:
Please use `&&` instead of `and` (and similar) throughout your code!

Reply:
done.